src/shogun/classifier/LDA.h

```diff
@@ -182,24 +182,21 @@ class CLDA : public CLinearMachine
 		 *
 		 * @return whether training was successful
 		 */
-		virtual bool train_machine(CFeatures* data=NULL);
-
-	   	/**
-		 * A templated specialization of the train_machine method
-		 * @param features training data
-		 * @param labels labels for training data
-		 * @see train_machine
-		 */
-		template <typename ST>
-		bool train_machine_templated();
+		template <typename ST, typename U = typename std::enable_if_t<std::is_floating_point<ST>::value>>
+		bool train_machine_templated(CDenseFeatures<ST>* data);
+		virtual bool support_dispatching()
```

Comment:
cant this sit inside the mixin?

Reply:
actually we dont need it :) will remove altogether

Reply:
I think we do to enter the branch in CMachine::train or?